File: src/main/java/thedarkcolour/futuremc/enchantment/EnchantHelper.java
package thedarkcolour.futuremc.enchantment;

import net.minecraft.enchantment.Enchantment;
import net.minecraft.enchantment.EnchantmentHelper;
import net.minecraft.item.ItemStack;

import java.util.Map;

public final class EnchantHelper {
    public EnchantHelper() {
    }

    public static boolean hasChanneling(ItemStack stack) {
        if(stack == null) {
            return false;
        }
        return EnchantmentHelper.getEnchantmentLevel(Enchantments.CONDUCTIVIDAD, stack) > 0;
    }

    public static int getImpaling(ItemStack stack) {
        if(stack == null) {
            return 0;
        }
        return EnchantmentHelper.getEnchantmentLevel(Enchantments.EMPALAMIENTO, stack);
    }

    public static int getLoyalty(ItemStack stack) {
        if(stack == null) {
            return 0;
        }
        return EnchantmentHelper.getEnchantmentLevel(Enchantments.LEALTAD, stack);
    }

    public static int getRiptide(ItemStack stack) {
        if(stack == null) {
            return 0;
        }
        return EnchantmentHelper.getEnchantmentLevel(Enchantments.RIPTIDE, stack);
    }

    public static int getQuickCharge(ItemStack stack) {
        if(stack == null) {
            return 0;
        }
        return EnchantmentHelper.getEnchantmentLevel(Enchantments.QUICK_CHARGE, stack);
    }

    public static boolean hasMultishot(ItemStack stack) {
        if(stack == null) {
            return false;
        }
        return EnchantmentHelper.getEnchantmentLevel(Enchantments.MULTISHOT, stack) > 0;
    }

    public static boolean isCursed(ItemStack stack) {
        if(stack.isEmpty()) return false;
        Map<Enchantment, Integer> map = EnchantmentHelper.getEnchantments(stack);
        for(Enchantment e : map.keySet()) {
            if(e.isCurse()) {
                return true;
            }
        }
        return false;
    }
}

File: src/main/java/thedarkcolour/futuremc/enchantment/EnchantHelper.kt
package thedarkcolour.futuremc.enchantment

import net.minecraft.enchantment.EnchantmentHelper
import net.minecraft.item.ItemStack
import thedarkcolour.futuremc.registry.FEnchantments

object EnchantHelper {
    fun getChanneling(stack: ItemStack): Boolean {
        return EnchantmentHelper.getEnchantmentLevel(FEnchantments.CONDUCTIVIDAD, stack) > 0
    }

    fun getImpaling(stack: ItemStack): Int {
        return EnchantmentHelper.getEnchantmentLevel(FEnchantments.EMPALAMIENTO, stack)
    }

    fun getLoyalty(stack: ItemStack): Int {
        return EnchantmentHelper.getEnchantmentLevel(FEnchantments.LEALTAD, stack)
    }

    fun getRiptide(stack: ItemStack): Int {
        return EnchantmentHelper.getEnchantmentLevel(FEnchantments.RIPTIDE, stack)
    }

    fun getQuickCharge(stack: ItemStack): Int {
        return EnchantmentHelper.getEnchantmentLevel(FEnchantments.QUICK_CHARGE, stack)
    }

    fun getMultishot(stack: ItemStack): Boolean {
        return EnchantmentHelper.getEnchantmentLevel(FEnchantments.MULTISHOT, stack) > 0
    }

    fun isCursed(stack: ItemStack): Boolean {
        if (stack.isEmpty) return false
        val map = EnchantmentHelper.getEnchantments(stack)
        for (e in map.keys) {
            if (e.isCurse) {
                return true
            }
        }
        return false
    }
}